MolecularNodes/md.py

```diff
@@ -61,16 +61,101 @@ def execute(self, context):
         context.scene.list_index = min(max(0, index - 1), len(my_list) - 1)
         
         return {'FINISHED'}
+    
+
+
+def selection_shell(
+                    file_top, 
+                    file_traj,
+                    solute ="element Li",
+                    solvent_list='resid 1-235,resid 235-600,byres element P',
+                    solvent_name ='EA,FEC,PF6',
+                    solvent_count = "3,1,0",
+                    solute_index_input=603, 
+                    frame_input=0,
+                    name = "default"
+                    ):
+
+
+    from solvation_analysis.solute import Solute
+    import MDAnalysis as mda
+
+    univ = mda.Universe(file_top, file_traj)
+    solute_ = univ.select_atoms(solute)
+
+    list_solvent = solvent_list.split(",")  
+
+    list_solv_name= solvent_name.split(",")
+
+    list_solvent_count = solvent_count.split(",")
+
+    solv_count = []
```

Comment:
I think there is a lot of clean-up that could be done in this function. Before I give advice on rewriting it, please comment this code to tell me what the different parts are accomplishing.